kura/examples/org.eclipse.kura.example.wire.math.singleport.provider/src/main/java/org/eclipse/kura/example/wire/math/singleport/RunningMedian.java

```diff
@@ -0,0 +1,35 @@
+package org.eclipse.kura.example.wire.math.singleport;
+
+import java.util.ArrayDeque;
+import java.util.Arrays;
+
+public class RunningMedian {
+
+    private final ArrayDeque<Double> window;
+    private final int windowSize;
+
+    public RunningMedian(final int windowSize) {
+        this.window = new ArrayDeque<>(windowSize);
+        this.windowSize = windowSize;
+    }
+
+    public double updateAndGetMedian(double value) {
+        int index = 0;
+        if (this.window.size() >= windowSize) {
+            this.window.removeFirst();
+        }
+        this.window.addLast(value);
+        double[] values = new double[windowSize];
+        for (Double windowValue : window) {
+            values[index++] = windowValue.doubleValue();
+        }
+        Arrays.sort(values);
+        if (windowSize % 2 == 0) {
```

Comment:
The buffer may not be at full capacity. Did you mean `window.size()`?

Reply:
noted.